src/tests/resistance/post_handshake/random_bytes.rs

```diff
@@ -0,0 +1,258 @@
+use bytes::{BufMut, BytesMut};
+use tempfile::TempDir;
+use tokio::time::sleep;
+
+use crate::{
+    protocol::codecs::{payload::Payload, tagmsg::Tag},
+    setup::node::Node,
+    tests::resistance::WAIT_FOR_DISCONNECT,
+    tools::{
+        constants::{
+            ERR_NODE_ADDR, ERR_NODE_BUILD, ERR_NODE_CONNECT, ERR_NODE_STOP, ERR_SYNTH_BUILD,
+            ERR_SYNTH_UNICAST, ERR_TEMPDIR_NEW,
+        },
+        synthetic_node::SyntheticNodeBuilder,
+        util::gen_rand_bytes,
+    },
+};
+
+/// Send given bytes directly to the node after the handshake and return the connection status.
+async fn send_bytes_to_the_node(data: Vec<u8>, debug: bool) -> bool {
+    // Spin up a node instance.
+    let target = TempDir::new().expect(ERR_TEMPDIR_NEW);
+    let mut node = Node::builder()
+        .log_to_stdout(debug)
+        .build(target.path())
+        .expect(ERR_NODE_BUILD);
+    node.start().await;
+
+    // Create a synthetic node and disable handshaking.
+    let synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect(ERR_SYNTH_BUILD);
+
+    let net_addr = node.net_addr().expect(ERR_NODE_ADDR);
+
+    // Create a connection without the handshake.
+    synthetic_node
+        .connect(net_addr)
+        .await
+        .expect(ERR_NODE_CONNECT);
+
+    // Send some random data.
+    let random_data_msg = Payload::RawBytes(data);
+    synthetic_node
+        .unicast(net_addr, random_data_msg)
+        .expect(ERR_SYNTH_UNICAST);
+
+    // Give some time to the node to kill our connection.
+    sleep(WAIT_FOR_DISCONNECT).await;
+
+    let is_connected = synthetic_node.is_connected(net_addr);
+
+    // Gracefully shut down the nodes.
+    synthetic_node.shut_down().await;
+    node.stop().expect(ERR_NODE_STOP);
+
+    is_connected
+}
+
+fn gen_tagged_msg_with_random_data(tag: Tag, len: usize) -> Vec<u8> {
+    let mut msg_content_random = BytesMut::from(Tag::get_tag_str(&tag).as_bytes());
```

Comment:
if we ultimately return a `Vec`, maybe we should start with one here (and then extend it) too?

Reply:
Done